app/services/capital_collator_and_assessor.rb

```diff
@@ -1,31 +1,45 @@
 class CapitalCollatorAndAssessor
+  PersonCapitalCollation = Struct.new(:total_vehicle)
+  CapitalCollation = Struct.new(:applicant_capital_collation, :partner_capital_collation)
   class << self
     def call(assessment)
-      data = Collators::CapitalCollator.call(
-        submission_date: assessment.submission_date,
-        capital_summary: assessment.capital_summary,
-        maximum_subject_matter_of_dispute_disregard: maximum_subject_matter_of_dispute_disregard(assessment),
-        pensioner_capital_disregard: pensioner_capital_disregard(assessment),
-      )
-      assessment.capital_summary.update!(data)
+      data = collate_applicant_capital(assessment)
+      assessment.capital_summary.update!(data.except(:total_vehicle))
```

Comment:
The functional spike seemed to suggest that the return of :total_vehicle is part of the problem. It hides the calculation, making that code do the calc twice. I think moving the calculation up one layer would solve the issue, as it would then be available, and its only sum(vehicles.map(&assessed_value)) so IMHO its ok to move

Reply:
I'm not sure I see the issue - we don't do the calculation twice, _because_ we return total vehicle. Yes, eventually that layer will _also_ return more information from each individual vehicle assessment calculation, required to populate additional parts of the payload, but I think that's going to be a case of more stuff being available in `data` here and being included in the object that's returned, as and when we do that refactor.